amqp-postgres/amqp_postgres/postgres_publisher.py

```diff
@@ -282,6 +282,18 @@ def _store_nobatch(self, conn, items):
             except (psycopg2.IntegrityError, ValueError):
                 logger.debug('Error storing %s: %s', exchange, item)
                 conn.rollback()
+            except psycopg2.ProgrammingError as e:
+                if e.pgcode == psycopg2.errorcodes.UNDEFINED_COLUMN:
```

Comment:
undefined column happens often on snap-restore, it's not really a surprising condition